### PR TITLE
Remove `fastai` from the dependencies

### PR DESCRIPTION
**Pull request by artificiala**
In this pull request, the library [fastai](https://docs.fast.ai/) is removed. 

The list below shows the functions from `fastai` used in the `pythainlp.ulmfit` module:

- BaseTokenizer -- an abstract class defining the basic functionality of Tokenizer used in the `fastai`
- TK_REP and TK_WREP -- constant values to represent repetition at the character level and word level respectively
- fix_html -- a function to replace HTML string. For example, replace 'amp;' with '&' in the text.
- replace_all_caps -- a function to replace all capitalized character by their lower version and add a token `TK_UP` before
- rm_useless_spaces -- a function to remove multiple space
- spec_add_spaces -- a function to add spaces around '/' and '#'. For example, it changes from "I #like to" to "I  # like to"

--------------------------------

New changes (Sat 31 Aug 2019):
1. `pythainlp/ulmfit/__init__.py`:
- Merge code from this commit https://github.com/PyThaiNLP/pythainlp/commit/75585650995a0af9b38d0082ccab1ed038295f35 (implemented by K. Charin). As a result, new text processing rules were added. Also, a new function `pythainlp.ulmfit.process_thai` that performs text processing to Thai text with a set of pre-defined rules such as replace repetitive words/characters with a special token or remove white space token from a list of tokens.

2.`tests/test_ulmfit.py`:

- add new test case: test_remove_space
- add new test case: test_replace_wrep_post_nonum
- add new test case: test_replace_wrep_post
- add new test case: test_replace_rep_nonum
- add new test case: test_replace_rep_after
- remove a test case: test_replace_all_caps (as the function `replace_all_caps` was removed from /pythianlp/ulmfit/__init__.py)

-------

Todos:

- [x] Add new test case for `pythainlp.ulmfit.process_thai` 
- [x] Add documentation for the following functions: 
    - [x] `pythainlp.ulmfit.process_thai` 
    - [x] `pythainlp.ulmfit.fix_html` 
    - [x] `pythainlp.ulmfit.replace_rep_after`
    - [x] `pythainlp.ulmfit.replace_rep_nonum`
    - [x] `pythainlp.ulmfit.replace_wrep_post`
    - [x] `pythainlp.ulmfit.replace_wrep_post_nonum`
    - [x] `pythainlp.ulmfit.remove_space`

### CODE REVIEW

**Comment by bact** [comment]
Looks like we have a problem with pytorch installation on Windows, so the AppVeyor build failed.

Please consider install pytorch using this command:
`pip install torch==1.2.0+cpu torchvision==0.4.0+cpu -f https://download.pytorch.org/whl/torch_stable.html`

See the new appveyor.yml here:
https://github.com/PyThaiNLP/pythainlp/blob/dev/appveyor.yml

**Comment by bact** [comment]
PyTorch problem sovled.
New issue is marisa-trie build on Windows + Python 3.7

**Comment by bact** [comment]
Build succeed on both Python 3.6 and Python 3.7.

File: pythainlp/benchmarks/word_tokenisation.py [hunk 1]
```diff
@@ -1,310 +0,0 @@
-# -*- coding: utf-8 -*-
```
Comment (wannaphongcom) [inline pythainlp/benchmarks/word_tokenisation.py]: - [ ] Add benchmarks

File: pythainlp/__init__.py [hunk 2]
```diff
@@ -34,4 +34,3 @@
 )
 from pythainlp.transliterate import romanize, transliterate
 from pythainlp.util import collate, thai_strftime
-from pythainlp.benchmarks import benchmark
```
Comment (wannaphongcom) [inline pythainlp/__init__.py]: - [ ] Add pythainlp.benchmarks

File: docs/api/benchmarks.rst [hunk 3]
```diff
@@ -1,24 +0,0 @@
-.. currentmodule:: pythainlp.benchmarks
```
Comment (wannaphongcom) [inline docs/api/benchmarks.rst]: - [ ] Add benchmarks docs

File: bin/word-tokenization-benchmark [hunk 4]
```diff
@@ -1,121 +0,0 @@
-#!/usr/bin/env python3
```
Comment (wannaphongcom) [inline bin/word-tokenization-benchmark]: - [ ] Add bin/word-tokenization-benchmark

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -36,8 +36,8 @@ We use the famous [gitflow](http://nvie.com/posts/a-successful-git-branching-mod
 
 # Discussion
 
-- Facebook group (for Thai NLP Discussion only): https://www.facebook.com/groups/thainlp
```
Comment (wannaphongcom) [inline CONTRIBUTING.md]: - [ ] Add Discussion

File: pythainlp/benchmarks/word_tokenisation.py (hunk 1 above)
Comment (artificiala) [reply pythainlp/benchmarks/word_tokenisation.py]: That was my mistake, I already fixed it. 😥

File: setup.py [hunk 6]
```diff
@@ -93,10 +90,7 @@
         "Topic :: Text Processing :: General",
         "Topic :: Text Processing :: Linguistic",
     ],
-    scripts=[
```
[the remark this answers is not in the document]
Comment (artificiala) [reply setup.py]: That was my mistake, I already fixed it. 😥

File: tests/data/sentences.yml [hunk 7]
```diff
@@ -1,42 +0,0 @@
-sentences:
```
[the remark this answers is not in the document]
Comment (artificiala) [reply tests/data/sentences.yml]: That was my mistake, I already fixed it. 😥

File: tests/test_benchmarks.py [hunk 8]
```diff
@@ -1,85 +0,0 @@
-import datetime
```
[the remark this answers is not in the document]
Comment (artificiala) [reply tests/test_benchmarks.py]: That was my mistake, I already fixed it. 😥

File: pythainlp/__init__.py (hunk 2 above)
Comment (artificiala) [reply pythainlp/__init__.py]: That was my mistake, I already fixed it. 😥

File: docs/api/benchmarks.rst (hunk 3 above)
Comment (artificiala) [reply docs/api/benchmarks.rst]: That was my mistake, I already fixed it. 😥

File: bin/word-tokenization-benchmark (hunk 4 above)
Comment (artificiala) [reply bin/word-tokenization-benchmark]: That was my mistake, I already fixed it. 😥

File: CONTRIBUTING.md (hunk 5 above)
Comment (artificiala) [reply CONTRIBUTING.md]: That was my mistake, I already fixed it. 😥

File: setup.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (artificiala) [reply setup.py]: Now for this PR, changes are only in 4 files.

- docs/api/ulmfit.rst
- pythainlp/ulmfit/__init__.py
- tests/test_ulmfit.py
- setup.py (only remove `fastai`)

**Comment by wannaphongcom** [comment]
remove fast.ai in setup.py (full) `"fastai>=1.0.38",`

**Comment by artificiala** [comment]
- [ x ] remove fastai in setup.py (full) "fastai>=1.0.38",
- [ x ] remove fastai in setup.py (ml) "fastai>=1.0.38",

**Pull request merged**